activerecord/lib/active_record/inheritance.rb

```diff
@@ -40,6 +40,8 @@ module Inheritance
       # This is true, by default.
       class_attribute :store_full_sti_class, instance_writer: false
       self.store_full_sti_class = true
+
+      @_type_candidates_cache = Concurrent::Map.new
```

Comment:
Needed this because tests is written using `ActiveRecord::Base`, for example [1](https://github.com/rails/rails/blob/master/activerecord/test/cases/inheritance_test.rb#L61), [2](https://github.com/rails/rails/blob/master/activerecord/test/cases/reflection_test.rb#L103). But I'm not sure if  it is right to call `compute_type` directly on `ActiveRecord::Base` maybe i should change the tests instead.